cardano-cli/src/Cardano/CLI/Ping.hs

```diff
@@ -0,0 +1,582 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.CLI.Ping
```

Comment:
this module should exist in `cardano-cli/src/Cardano/CLI/Shelley/Run`

Reply:
I put the code a level higher because the command is not Shelley specific.